pkg/serializer/serializer.go

```diff
@@ -163,22 +184,55 @@ func (s Serializer) serializePayload(payload marshaler.Marshaler, compress bool,
 	return payloads, extraHeaders, nil
 }
 
-func (s Serializer) serializeStreamablePayload(payload marshaler.StreamJSONMarshaler) (forwarder.Payloads, http.Header, error) {
-	payloads, err := s.seriesPayloadBuilder.Build(payload)
+func (s Serializer) serializeStreamablePayload(payload marshaler.StreamJSONMarshaler, policy jsonstream.OnErrItemTooBigPolicy) (forwarder.Payloads, http.Header, error) {
+	payloads, err := s.seriesPayloadBuilder.BuildWithOnErrItemTooBigPolicy(payload, policy)
 	return payloads, jsonExtraHeadersWithCompression, err
 }
 
+func (s Serializer) serializeEventsStreamJSONMarshalerPayload(
```

Comment:
nit: A comment with the logic written in plain english might make sense here, it would be hard for someone external to understand what's going on IMO

Reply:
I tend to use comments:
 - To explain the rationales of why the code is written in a specific way
- When the code is tricky and cannot be simplified.
I tend to avoid comments when they do not bring more value than reading the code as comments become quickly out dated.

Anyway, I added a plain english comment.

Reply:
Thanks